airflow/hooks/dbapi.py

```diff
@@ -53,6 +53,8 @@ class DbApiHook(BaseHook):
     supports_autocommit = False
     # Override with the object that exposes the connect method
     connector = None  # type: Optional[ConnectorProtocol]
+    # Override to use different schema from connection setting
+    schema = None
```

Comment:
I believe the schema here is actually an instance attribute so it should go inside the `__init__` call.

Reply:
This is better IMO, because a subclass can simply do `self.schema = None` in its `__init__` without worrying the value getting overridden by `super().__init__`.

Reply:
> self.schema = None in its __init__ without worrying the value getting overridden by super().__init__.

Hmm, shouldn't `super().__init__` being ran before other subclass-specific params? Also, all other superclass shouldn't worry about `self.schema = <schema>` because it should be captured by `super().__init__()`
```python
def __init__(self, *args, **kwargs):
    super.__init__(*args, **kwargs)    # `schema` should go in here
```

I think it's rather confusing right now because we're mixing class attribute and instance attribute. What's the purpose of `schema` at the class level? When we do `self.schema = None` in subclass, the class attribture `schema` will simply get ignored and hidden by the instance attibute `self.schema`. i.e.

```python
class Parent:
    schema = None


class Child(Parent):

    def __init__(self, schema):
        self.schema = schema


child = Child('postgres')

print(child.__class__.schema)
print(child.schema)
print(Child.schema)
print(Parent.schema)

# None
# postgres
# None
# None
```

Reply:
Ah I see, you mean we should add `schema` as an argument on `DbApiHook`, and use it to populate `self.schema`. Yes, that would be a reasonable design.

Previously I thought you were suggesting something like

```python
class DbApiHook:
    def __init__(self, ...):  # No schema= here.
        ...
        self.schema = None

class PostgresHook(DbApiHook):
    def __init__(self, ..., schema=None):
        super().__init__(...)
        self.schema = schema
```

This would be more difficult to maintain (not *too* difficult, but comparitively so).

Reply:
Ahh, no lol. Yeah, I'm suggesting adding it as an argument to `DbApiHook` and pass it through the `__init__` call, so all subclasses can all inherit this properly.

Reply:
I've made some change at latest commit:

1. Move schema to instance attribute in DbApiHook
2. Set `self.schema` from kwargs or None in `DbApiHook.__init__`
3. Remove the code same as (2) in `PostgresHook.__init__`

I think those are what you discussed above?